src/components/navigation/document-tab-panel.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+import { BaseComponent, IBaseProps } from "../base";
+import { Tab, Tabs, TabList, TabPanel } from "react-tabs";
+import { ELeftTabSectionType, LeftTabSpec, LeftTabSectionModelType } from "../../models/view/left-tabs";
+import { IStores } from "../../models/stores/stores";
+import { TabPanelDocumentsSection } from "../thumbnail/tab-panel-documents-section";
+import { DocumentDragKey, DocumentModelType, SupportPublication } from "../../models/document/document";
+import { LogEventName, Logger } from "../../lib/logger";
+import { EditableDocumentContent } from "../document/editable-document-content";
+
+import "./document-tab-panel.sass";
+
+interface IProps extends IBaseProps {
+  tabSpec: LeftTabSpec;
+  onDocumentClick?: (document: DocumentModelType) => void;
+  onTabClick?: () => void;
+  document?: DocumentModelType;
+}
+
+interface IState {
+  tabIndex: number;
+}
+
+const kNavItemScale = 0.11;
+
+@inject("stores")
+@observer
+export class DocumentTabPanel extends BaseComponent<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      tabIndex: 0
+    };
+  }
+
+  public render() {
+    const { document, tabSpec, onTabClick, onDocumentClick } = this.props;
+    const { tabIndex } = this.state;
+    const { user } = this.stores;
+    const leftTabSpecs = this.stores.appConfig.leftTabs.tabSpecs;
+    const leftTabSpec = leftTabSpecs.find(spec => spec.tab === tabSpec.tab);
+    return (
+      <Tabs
+        className={`document-tabs ${leftTabSpec?.tab}`}
+        forceRenderTabPanel={true}
+        onSelect={this.handleTabSelect}
+        selectedIndex={tabIndex}
+        selectedTabClassName="selected"
+      >
+        <TabList className={`tab-list ${leftTabSpec?.tab}`} onClick={onTabClick}>
+          {leftTabSpec?.sections.map((section) => {
+            const sectionTitle = this.getSectionTitle(section, this.stores);
+            return (
+              <Tab
+                className={`doc-tab ${leftTabSpec?.tab} ${section.type}`}
+                key={`section-${section.type}`}
+                data-test={section.dataTestHeader}
+              >
+                {sectionTitle}
+              </Tab>
+            );
+          })}
+        </TabList>
+        {leftTabSpec?.sections.map((section, index) => {
+          const _handleDocumentStarClick = section.showStarsForUser(user)
+                ? this.handleDocumentStarClick
+                : undefined;
+          const _handleDocumentDeleteClick = section.showDeleteForUser(user)
+                ? this.handleDocumentDeleteClick
+                : undefined;
+          return (
+            <TabPanel key={`section-${section.type}`}>
+              { document && (index === tabIndex)
+                ? <EditableDocumentContent
+                    mode={"1-up"}
+                    isPrimary={false}
+                    document={document}
+                    readOnly={true}
+                  />
```

Comment:
For some reason that I can't quite articulate, this embedded use of `<EditableDocumentContent>` bothers me. What if instead this component took an optional `DocumentViewer` prop and this was changed to:
```typescript
{ document && DocumentViewer && (index === tabIndex)
  ? <DocumentViewer document={document} />
  : <TabPanelDocumentsSection ...
```
Then on my side I would leave the prop blank and on your side you would have:
```typescript
const RefDocumentViewer: React.FC<IRefDocViewerProps> = ({ document }) => {
  return <EditableDocumentContent mode="1-up" isPrimary={false} readOnly={true} />
};
...
<DocumentTabPanel DocumentViewer={RefDocumentViewer} ... />
```
```

Reply:
Or maybe even better, since the `document` is a prop as well, we could combine both into a single `documentView` property and then you would pass `<DocumentTabPanel docView=<EditableDocumentContent document={document} ... />` and I would leave it blank.

Reply:
I agree.  It feels weird, and I wasn't entirely pleased with jamming `EditableDocumentContent` in there.  I like the idea of passing the component in.

Reply:
I went the route of your second comment.  I like this better.  `DocumentTabPanel` doesn't feel so confusing (especially once we strip those handlers out).